scipy/stats/tests/test_stats.py

```diff
@@ -383,10 +388,9 @@ def test_pROUNDROUND(self):
         assert_approx_equal(r,1.0)
 
 
-@make_skip_xp_backends('pearsonr')
+@make_skip_xp_backends(stats.pearsonr)
 class TestPearsonr:
-    @skip_xp_backends(np_only=True)
```

Comment:
I got carried away here cleaning up, I can revert if desired!

Reply:
I was a little concerned by the size of the diff but it's not too bad.